### PR TITLE
chore: upgrade and improve date/time picker

### PR DESCRIPTION
**Pull request by mh-alahdadian**

### CODE REVIEW

File: packages/mui/src/constant-types/constant-types.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+   * For Persian Language & Jalaali Calender
+   */
+  fa = 'fa',
+  /**
+   * For English Language & Gregorian Calender
+   */
+  en = 'en',
+}
+
+export namespace Locale {
```
Comment (mdeljavan) [inline packages/mui/src/constant-types/constant-types.ts:16]: This file is not used, so delete it safely.

File: packages/mui/src/pickers-common/picker-props.tsx [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+  DateTimePickerProps,
+  MuiPickersAdapterContext,
+} from '@mui/x-date-pickers';
+import { DateTimeValidationError } from '@mui/x-date-pickers/internals/hooks/validation/useDateTimeValidation';
+import { ReactNode, useContext, useState } from 'react';
 
-import { PickerLabel, usePickerI18nContext } from './px-picker-i18n-provider';
-
-type RenderInput = DateTimePickerProps['renderInput'];
+type RenderInput = DateTimePickerProps<any, any>['renderInput'];
```
Comment (mdeljavan) [inline packages/mui/src/pickers-common/picker-props.tsx]:
```suggestion
type RenderInput = DateTimePickerProps<unknown, unknown>['renderInput'];
```

File: packages/mui/src/pickers-common/picker-props.tsx [hunk 3]
```diff
@@ -24,26 +26,18 @@ export const pickerDefaultRenderInputFactory =
       />
     );
 
-export function usePickerProps(props: { errorsText: PickerLabel['errors'] }) {
-  const contextLabel = usePickerI18nContext();
-
+export function usePickerProps<In, Out>(props?: {}) {
```
Comment (mdeljavan) [inline packages/mui/src/pickers-common/picker-props.tsx]: Props is not used

File: packages/mui/src/pickers-common/picker-props.tsx (hunk 2 above)
Comment (mh-alahdadian) [reply packages/mui/src/pickers-common/picker-props.tsx]: done

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**